src/Microsoft.OpenApi/Writers/SpecialCharacterStringExtensions.cs

```diff
@@ -188,7 +189,7 @@ internal static string GetYamlCompatibleString(this string input)
 
             // If string can be mistaken as a number, a boolean, or a timestamp,
             // wrap it in quote to indicate that this is indeed a string, not a number, a boolean, or a timestamp
-            if (decimal.TryParse(input, out var _) ||
+            if (decimal.TryParse(input, NumberStyles.Any, CultureInfo.InvariantCulture, out var _) ||
```

Comment:
>NumberStyles.Any [](start = 40, length = 16)

This can be a bit stricter. Could you change this to Float? 

JSON/YAML only treats what roughly corresponds to NumberStyles.Float in C# as a number.

Reply:
Resolved.